src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -233,7 +241,7 @@ private TensorFlowTransform(IHostEnvironment env, byte[] modelBytes, string[] in
             {
                 var tfOutput = new TFOutput(Graph[Outputs[i]]);
                 var shape = Graph.GetTensorShape(tfOutput);
-                int[] dims = shape.ToIntArray().Skip(shape[0] == -1 ? BatchSize : 0).ToArray();
+                int[] dims = shape.NumDimensions > 0 ? shape.ToIntArray().Skip(shape[0] == -1 ? BatchSize : 0).ToArray() : new[] { 0 };
```

Comment:
>shape.NumDimensions > 0 [](start = 29, length = 23)

I am presuming we are using this here as a check for models producing variable length outputs...Am i right ?

Does TF document such behaviour somewhere ?

Reply:
If the shape is unknown, it has shape.NumDimensions == -1, and shape.ToIntArray() == null, which would cause a null reference exception when we try to access shape[0].

---
In reply to: [216462762](https://github.com/dotnet/machinelearning/pull/857#discussion_r216462762) [](ancestors = 216462762)